### PR TITLE
Production style docker environment (fixes #52)

### LINKED ISSUE #52: Create a docker container
We should dockerise this so we can have an easy way of standing up an instance. We should also include this in the olympia docker-compose.

### PR DESCRIPTION
**Pull request by mstriemer**
I'm not sure of the best way to get the dev server to reload code so this is running the "production" server.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM node:argon
```
Comment (muffinresearch) [inline Dockerfile]: This should be based on centos like addons-server. Also node should come from our own builds which are mirrored from nodesource.

See https://bugzilla.mozilla.org/show_bug.cgi?id=1226533 for info.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ADD package.json /srv/addons-frontend-node
+WORKDIR /srv/addons-frontend-node
+RUN npm install -g npm@3
+RUN npm install
+
+ADD . /code
+WORKDIR /code
+
+# Replace the local node_modules with the ones we installed above.
+RUN rm -rf node_modules
```
Comment (muffinresearch) [inline Dockerfile]: I see - this won't typically be needed in a docker hub build but you need this when building from your local src.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+FROM node:argon
+
+# Install node_modules into a different directory to avoid npm/npm#9863.
+RUN mkdir -p /srv/addons-frontend-node
+ADD package.json /srv/addons-frontend-node
+WORKDIR /srv/addons-frontend-node
+RUN npm install -g npm@3
+RUN npm install
+
+ADD . /code
```
Comment (muffinresearch) [inline Dockerfile]: I think we should start using /srv for everything since /code is not a standard dir.

File: Dockerfile (hunk 1 above)
Comment (mstriemer) [inline Dockerfile]: Is that just so that it mimics production?

Comment (muffinresearch) [inline Dockerfile]: Yeah it makes to try and build everything rougly similar unless there's a good reason not to.

**Comment by mstriemer** [comment]
I updated it to use `centos:centos7` and fetch node from nodesource. How do we set this up on docker hub? I'm guessing an admin has to do that?

**Comment by muffinresearch** [comment]
@mstriemer I can add you to the group if you're not already on it. It's pretty straight-forward. It'll re-build on updates to master.

**Comment by muffinresearch** [comment]
r+ nice!

**Pull request merged**